### PR TITLE
`azurerm_mssql_virtual_machine` - support `sql_instance_setting`

### LINKED ISSUE #15824: Support for instance settings on azurerm_mssql_virtual_machine
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Option to supply SQL Instance_settings config block when using the azurerm_mssql_virtual_machine resource. When deploying a SQL Server VM in the portal, we have the option to set storage configuration which is supported in the resource. However, we can also set Instance settings which is not supported in the resource. These are the missing settings:

- Server Collation (string)
- MAXDOP (number)
- Minimum server memory (MB) (number)
- Maximum server memory (MB) (number)
- Optimize for ad-hoc workload (bool)

### New or Affected Resource(s)

* azurerm_mssql_virtual_machine

### Potential Terraform Configuration

```hcl
resource "azurerm_mssql_virtual_machine" "example" {
  virtual_machine_id               = data.azurerm_virtual_machine.example.id
  sql_license_type                 = "PAYG"
  r_services_enabled               = true
  sql_connectivity_port            = 1433
  sql_connectivity_type            = "PRIVATE"
  sql_connectivity_update_password = "Password1234!"
  sql_connectivity_update_username = "sqllogin"

  auto_patching {
    day_of_week                            = "Sunday"
    maintenance_window_duration_in_minutes = 60
    maintenance_window_starting_hour       = 2
  }

  instance_settings {
    collation         = "SQL_Latin1_General_CP1_CI_AS"
    maxdop            = 0
    min_server_memory = 0
    max_server_memory = 2147483647
    adhoc_workload    = false
  }
}
```

### PR DESCRIPTION
**Pull request by myc2h6o**
Fix #15824

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result
image

File: internal/services/mssql/mssql_virtual_machine_resource.go [hunk 1]
```diff
@@ -315,6 +315,64 @@ func resourceMsSqlVirtualMachine() *pluginsdk.Resource {
 				ValidateFunc: validate.SqlVirtualMachineLoginUserName,
 			},
 
+			"sql_instance_setting": {
```
Comment (katbyte) [inline internal/services/mssql/mssql_virtual_machine_resource.go:318]: setting seems redundant?
```suggestion
			"sql_instance": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
just one comment to address and then this looks good

File: internal/services/mssql/mssql_virtual_machine_resource.go (hunk 1 above)
Comment (myc2h6o) [reply internal/services/mssql/mssql_virtual_machine_resource.go]: Have removed `_setting`.

**Comment by myc2h6o** [comment]
Hi @katbyte thanks for reviewing the pr! I've updated the code according to the comment. Please take a look

**Review by katbyte — APPROVED** [review]
thanks @myc2h6o - LGTM 🦀

**Pull request merged**